client/src/app/pages/courses/save/save-course.component.ts

```diff
@@ -1,31 +1,33 @@
 import { Component, OnInit, OnDestroy } from '@angular/core';
 import { Router, ActivatedRoute, Params } from '@angular/router';
+import { Subscription } from 'rxjs';
 import { Course } from 'app/shared/models/course.model';
 import { DurationPipe } from 'app/shared/pipes';
 import { CourseService } from 'app/services';
-import { Subscription } from 'rxjs';
 
 @Component({
     selector: 'save-course',
     templateUrl: './save-course.component.html',
     styleUrls: [ './save-course.styles.scss' ]
   })
   export class SaveCourseComponent implements OnInit, OnDestroy {
-    public course: Course;
+    public course: Course = {} as Course;
     public subscription: Subscription;
 
     constructor(
       private route: ActivatedRoute,
       private router: Router,
       private courseService: CourseService
-    ) {}
+    ) { }
 
     public ngOnInit() {
-      let id: string = this.route.params['id'];
+      let id: string;
+      this.route.params.subscribe((data) => {
+        id = data['id'];
+      });
 
       if (typeof id === 'undefined' || id === null) {
```

Comment:
and what if `this.route.params` resolved asynchronously, your code would break. You'd rather put everything into subscription callback.